### PR TITLE
project day 2

### PR DESCRIPTION
**Pull request by Jhillega**
my work for day 2

### CODE REVIEW

File: Arrays/Arrays/Array/Array/Tester.swift [hunk 1]
```diff
@@ -57,22 +87,92 @@ class Tester {
...
+            if array[compareIndex] == emptyChair{
+                emptySeats += 1
+                
+            }
+            else{
+                
+                fullSeats += 1
+            }
+        }
+        return (fullSeats - emptySeats) + 1
```
Comment (LukeRave) [inline Arrays/Arrays/Array/Array/Tester.swift:109]: `(fullSeats - emptySeats) + 1` may not work for all cases, try `["I", "O", "I", "I", "I"]` should return 3

File: Arrays/Arrays/Array/Array/Tester.swift [hunk 2]
```diff
@@ -12,18 +12,30 @@ class Tester {
...
 
     /// 2. Write a function to return the last half of the array, not including the median
     func lastHalf(array: [Int]) -> [Int] {
-        return []
+        let toPlay = array
+        let size = array.count
+        let half = size/2 + 1
+        let back = Array(toPlay[half...])
+        
+        return back
```
Comment (LukeRave) [inline Arrays/Arrays/Array/Array/Tester.swift:28]: `half` will be work for odd length arrays, may not work for a length of 2: [1,2]

Comment (Jhillega) [reply Arrays/Arrays/Array/Array/Tester.swift:35]: I made some changes, thanks for the feedback. Everything helps

File: Arrays/Arrays/Array/Array/Tester.swift [hunk 3]
```diff
@@ -57,22 +94,94 @@ class Tester {
...
+                
+            }
+            else{
+                
+                fullSeats += 1
+            }
+        }
+        //is the test correct? shouldn't it be 2?
+        let lukeModifier = 1
+        return (fullSeats - emptySeats) + lukeModifier
```
Comment (LukeRave) [inline Arrays/Arrays/Array/Array/Tester.swift:118]: If the input is [ I I I O O O] three people enters the room, then three people leaves the room, return should be 3.
The first 3 people all need chairs, so 3 was the num chairs needed, even after they leave, 3 was required at one point.

File: Arrays/Arrays/Array/Array/Tester.swift [hunk 4]
```diff
@@ -57,22 +94,94 @@ class Tester {
...
         
-        return 0
+        let size = array.count
+        let start = array.startIndex
+        
+        for num in 0 ... (size - 2) {
+            
+            let buyNum = array.index(start, offsetBy: num)
+            let sellNum = array.index(start, offsetBy: num + 1)
+
```
Comment (LukeRave) [inline Arrays/Arrays/Array/Array/Tester.swift:174]: no need to use startIndex and offsetBy with arrays, buyNum and sellNum can be ints

Comment (LukeRave) [reply Arrays/Arrays/Array/Array/Tester.swift:174]: Unless you're just flexing the use of Index/offsetBy, which is also cool

**Pull request merged**